cachito/workers/tasks/pip.py

```diff
@@ -17,3 +35,61 @@ def cleanup_pip_request(request_id):
         "username": get_hosted_repositories_username(request_id),
     }
     nexus.execute_script("pip_cleanup", payload)
+
+
+@app.task
+def fetch_pip_source(request_id, package_configs=None):
+    """
+    Resolve and fetch pip dependencies for a given request.
+
+    :param int request_id: the Cachito request ID this is for
+    :param list package_configs: the list of optional package configurations submitted by the user
+    """
+    validate_pip_config()
+    bundle_dir = RequestBundleDir(request_id)
+
+    log.info("Configuring Nexus for pip for the request %d", request_id)
+    set_request_state(request_id, "in_progress", "Configuring Nexus for pip")
+    pip_repo_name = get_pypi_hosted_repo_name(request_id)
+    raw_repo_name = get_raw_hosted_repo_name(request_id)
+    prepare_nexus_for_pip_request(pip_repo_name, raw_repo_name)
+
+    log.info("Fetching dependencies for request %d", request_id)
+    package_configs = package_configs or [{}]
+    packages_data = []
+    for pkg_cfg in package_configs:
+        pkg_path = pkg_cfg.get("path", ".")
+        source_dir = bundle_dir.app_subpath(pkg_path).source_dir
+        request = set_request_state(
+            request_id, "in_progress", f"Fetching dependencies at the {pkg_path!r} directory",
+        )
+        pkg_and_deps_info = resolve_pip(
+            source_dir,
+            request,
+            requirement_files=pkg_cfg.get("requirements_files"),
+            build_requirement_files=pkg_cfg.get("requirements_build_files"),
+        )
+
+        # defer DB operations to use the Nexus password in the env vars
+        packages_data.append(pkg_and_deps_info)
+
+    log.info("Finalizing the Nexus configuration for pip for the request %d", request_id)
+    set_request_state(request_id, "in_progress", "Finalizing the Nexus configuration for pip")
+    username = get_hosted_repositories_username(request_id)
+    password = finalize_nexus_for_pip_request(pip_repo_name, raw_repo_name, username)
+
+    # Set environment variables
+    raw_url = get_pypi_hosted_repo_url(request_id)
+    pip_index_url = get_index_url(raw_url, username, password)
+    env_vars = {"PIP_INDEX_URL": {"value": pip_index_url, "kind": "literal"}}
```

Comment:
When testing locally, I realized that with the `DevelopmentConfig`, this URL will not be accessible externally:
```
$ curl localhost:8080/api/v1/requests/1/environment-variables
{
  "PIP_INDEX_URL": {
    "kind": "literal", 
    "value": "http://cachito-pip-1:2f072c2325fa53d2299259004caf980d@nexus:8081/repository/cachito-pip-hosted-1/"
  }
}
```
nexus:8081 would have to be localhost:8082 instead

I assume this will only be an issue for the dev environment, not for any real environment. Is this something we need to worry about?

Reply:
This should come from `cachito_nexus_url`. In production environments they should contain a fqdn, so this would not be an issue in such environments.